[LiveComponent] Document how to use `extra_options` in the Ajax-powered autocomplete (#1702)
symfony/ux: src/Autocomplete/doc/index.rst

> Congratulations! Your ``EntityType`` is now Ajax-powered!

Reviewer: Just before there is a caution block concerning options ... maybe it could be updated / rewritten to push the extra option handling  ?

.. _passing-extra-options-to-the-ajax-powered-autocomplete:

> Passing Extra Options to the Ajax-powered Autocomplete

Reviewer: I'd move this block after the standard options.. wdyt ? 

(especially if we rewrite the caution block to add a link to this section)

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> .. versionadded:: 2.13

Reviewer: ```suggestion
.. versionadded:: 2.14
```

>     The ability to pass extra options was added in LiveComponents 2.13.

Reviewer: ```suggestion
    The ability to pass extra options was added in Autocomplete 2.14.
```

By default, when you pass any options while adding a field to a form type, they will be lost
when the autocomplete field is rendered on an Ajax call. To partially avoid this limitation,
the `extra_options` option was added. It can only hold a scalar values, but it covers most use cases.

Considering the following example, when the form type is rendered for the first time, it will use the `query_builder` defined
while adding a `food` field to the `FoodForm`. However, when the Ajax is used to fetch the results, on the consequent renders,

> the default `query_builder` will be used::

Reviewer: Just a personal feeling (and my english is quite limited so take this with a pinch of salt :) )

And i think @javiereguiluz vision would be much usefull than mine there :) 

What i feel here is that those two paragraphs may be written with a tone more "neutral / factual" (not sure of the term) 

Something like 

```
Autocomplete field options are not kept from a request to another, so following renders ... 

To persists options betwen [..], you must use extra_options [..]

::warning
only scalar (string, booleans ....) 
```

And then the example. 

not sure if what i'm saying is clear 👼

                    'query_builder' => function (EntityRepository $er) {
                            $qb = $er->createQueryBuilder('o');

                            $qb->andWhere($qb->expr()->notIn('o.id', [$currentFoodId]));

                            return $qb;
                        };
                    }
                ])
            ;
        }
    }

If some food can be consisted of other foods, we might want to exclude the "root" food from the list of available foods.
To achieve this, we can remove the `query_builder` option from the above example and pass the `excluded_foods` extra option
to the `FoodAutocompleteField`::

    // src/Form/FoodForm.php
    // ...

    class FoodForm extends AbstractType
    {
        public function buildForm(FormBuilderInterface $builder, array $options): void
        {
            $currentFoodId = $builder->getData()->getId();

            $builder
                ->add('food', FoodAutocompleteField::class, [
                    'extra_options' => [
                        'excluded_foods' => [$currentFoodId],
                    ],
                )
            ;
        }
    }

> The magic of the `extra_options` option is that it will be passed to the `FoodAutocompleteField` every time an Ajax call is made.

Reviewer: ```suggestion
The magic of the `extra_options` is that it will be passed to the `FoodAutocompleteField` every time an Ajax call is made.
```

So now, we can just use the `excluded_foods` extra option in the default `query_builder` of the `FoodAutocompleteField`::

    // src/Form/FoodAutocompleteField.php
    // ...

    use Symfony\Bundle\SecurityBundle\Security;
    use Symfony\UX\Autocomplete\Form\AsEntityAutocompleteField;
    use Symfony\UX\Autocomplete\Form\BaseEntityAutocompleteType;

    #[AsEntityAutocompleteField]
    class FoodAutocompleteField extends AbstractType
    {

>         public function configureOptions(OptionsResolver $resolver)

Reviewer: ```suggestion
        public function configureOptions(OptionsResolver $resolver): void
```

        {
            $resolver->setDefaults([
                // ...
                'query_builder' => function (Options $options) {
                    return function (EntityRepository $er) use ($options) {
                        $qb = $er->createQueryBuilder('o');

                        $excludedFoods = $options['extra_options']['excluded_foods'] ?? [];
                        if ([] !== $excludedFoods) {
                            $qb->andWhere($qb->expr()->notIn('o.id', $excludedFoods));
                        }

                        return $qb;
                    };

>                 }

Reviewer: I love this example because it's very relatable. 

On the other side, i'm not sure we want to push that much the "query_builder" option, as it may be hard to grasp for newcomers 

Just wondering if we could illustrate the extra_options with something "simpler" ... 
.. if not this example is great :)

    Set to ``focus`` to call the ``load`` function when control receives focus.
    Set to ``true`` to call the ``load`` upon control initialization (with an empty search).

``extra_options`` (default ``[]``)

>     Allow you to pass extra options for Ajax-based autocomplete fields.

Reviewer: Add a link to the related section ?

Author: I'm not sure if I'm 100% sure what you want to do :D? Can you show me an example snippet?

Reviewer: Me neither right now, so .... 😆

Usually, you'll pass this URL to the Stimulus controller, which is
discussed in the next section.

Passing Extra Options to the Autocompleter
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> .. versionadded:: 2.13

Reviewer: ```suggestion
.. versionadded:: 2.14
```

>     The ability to pass extra options was added in LiveComponents 2.13.

Reviewer: ```suggestion
    The ability to pass extra options was added in Autocomplete 2.14.
```
